.github/workflows/native-ubuntu.yml

```diff
@@ -0,0 +1,146 @@
+      with:
+        submodules: recursive
+
+    # The following packages are already installed on GitHub-hosted runners:
+    # build-essential openssl libssl-dev
+    # No need to install libprotobuf{17,10,9v5} on Ubuntu {20,18,16}.04 because
+    # it is installed together with libprotobuf-dev
+    - name: Install dependencies
+      run: |
+        sudo apt-get update 
+        sudo apt-get install -y \
+            libgoogle-perftools-dev libprotobuf-dev protobuf-compiler  \
+            libboost-all-dev g++-${{ matrix.gcc }} 
+
+    # https://software.intel.com/content/www/us/en/develop/articles/installing-intel-free-libs-and-python-apt-repo.html
+    - name: Install MKL
+      run: |
+        wget -qO- "https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS-2019.PUB" | sudo apt-key add -
+        sudo sh -c "echo deb https://apt.repos.intel.com/mkl all main > /etc/apt/sources.list.d/intel-mkl.list"
+        sudo apt-get update -o Dir::Etc::sourcelist="/etc/apt/sources.list.d/intel-mkl.list"
+        sudo apt-get install -y --no-install-recommends intel-mkl-64bit-2020.0-088
+      if: matrix.cmake.USE_WASM_COMPATIBLE_SOURCE == 'OFF'
+
+    # The script simplifies installation of different versions of CUDA
+    - name: Install CUDA
+      run: ./3rd_party/marian-dev/scripts/ci/install_cuda_ubuntu.sh ${{ matrix.cuda }}
+      if: matrix.gpu == 'ON'
+
+    # Boost is installed on GitHub-hosted runners in a non-standard location
+    # https://github.com/actions/virtual-environments/issues/687#issuecomment-610471671
+    - name: Configure CMake
+      run: |
+        mkdir -p build
+        cd build
+        CC=/usr/bin/gcc-${{ matrix.gcc }} CXX=/usr/bin/g++-${{ matrix.gcc }} CUDAHOSTCXX=/usr/bin/g++-${{ matrix.gcc }} \
+        cmake .. \
+          -DCMAKE_BUILD_TYPE=${{ matrix.cmake.CMAKE_BUILD_TYPE }}\
+          -DCOMPILE_TESTS=${{ matrix.cmake.COMPILE_TESTS }}\
+          -DCOMPILE_EXAMPLES=${{ matrix.cmake.COMPILE_EXAMPLES }} \
+          -DCOMPILE_SERVER=${{ matrix.cmake.COMPILE_SERVER }} \
+          -DUSE_WASM_COMPATIBLE_SOURCE=${{ matrix.cmake.USE_WASM_COMPATIBLE_SOURCE }} \
+
+    - name: Compile bergamot-translator
+      working-directory: build
+      run: make -j2
+
+    - name: Run unit tests
+      working-directory: build
+      run: make test
+      # GitHub-hosted VMs do not have GPUs, so can not be run in CUDA builds
+      if: matrix.gpu == 'OFF' && matrix.cmake.COMPILE_TESTS == 'ON'
+
+    - name: Print versions
+      working-directory: build
+      run: |
+        ./app/bergamot-translator-app --version
+
+
+    # Configure hooks to upload artifacts if the build is Slim
+    - name: Upload build artifacts
+      uses: actions/upload-artifact@v2
+      with: 
+        name: bergamot-translator-artifacts
+        path: |
+            ./build/app/service-cli*
+            ./build/app/bergamot-translator-app*
+            ./build/app/marian-decoder-new
+      if: matrix.cmake.CMAKE_BUILD_TYPE == 'Slim'
```

Comment:
I understood now. Regression tests are running for "full-marian-slim" only. Right?

Reply:
No, they're running for everything. It is only artifact upload is reduced to 'slim' (I don't want to upload big binaries with debug symbols). Not that since regression-tests are now in the same matrix entry, I don't need to upload for regression-tests.

This is more a future hook not complete, we can cut this out if it's causing any harm (my plans are to ship executables compatible with multiple OS (FYI, @kpu). Maybe a REST server that can be downloaded directly and used in user-testing to compare with WASM single thread as well?

Reply:
I think I am not clear then what is happening 😄

I was under the impression that uploading build artifacts step has something to do with running regression-tests. But looks like it is not?

Reply:
I'm confused too.  That demo about a multiplatform executable from somebody on the Internet was fun and all, but what does it have to do with the regression tests?

Reply:
So, when I set to Slim in a two workflow mode, it was suggested we maintain Release. The problem with Release are really big binaries which I don't want to move around and long-compile times.

1. Solve the big binary problem by keeping build and regression-test as steps in the same file. I have two builds here, Release and Slim now. We no longer require upload artifacts for the binaries, they're available in the same workflow-run.
2. Now we have growing list of files per workflow and inconsistencies between them. As a lazy programmer, I don't like this. I pick common parameters and collapse them into GitHub CI Matrix. While I'm at this, I see I can improve maintainability and have regression-tests in native as well with minimum effort (there's only one place which requires edits). So I collapse native into the same file as well. 
3. I choose to have some upload-artifacts code left from before, than cut it off, I keep it seeing the possibilities. Not that it immediately needs to be done, just putting out the possibilities of leaving it around. That is the upload code we're commenting on.

Reply:
Okay. In this case, my suggestion would be to remove the uploading artifacts step from both of these files if it is not serving any purpose right now. It is a super small code that can be added later as and when it is needed. We will keep the confusion away this way. However, I leave the decision up to you 👍

Reply:
I guess I'll remove it to avoid confusion for now. :D